File: problems/HackerRank/week4/pickingNumbers.py
def pickingNumbers(a):
    window_start = 0
    ans = 0
    a.sort()
    
    for window_end in range(1, len(a)):
        if a[window_end] - a[window_start] <= 1:
            ans = max(ans, window_end - window_start + 1)
        while a[window_end] - a[window_start] > 1:
            window_start += 1
            
    return ans


def pickingNumbers2(a):
    freq = [0] * 101
    
    for num in a:
        freq[num] += 1
        
    return max(freq[i] + freq[i+1] for i in range(100))

File: problems/HackerRank/Algorithms/Implementation/pickingNumbers.ts
function pickingNumbers(a: number[]): number {
  // sort arr in ascding order
  a.sort((x, y) => x - y);

  let ans = 0;
  let window_start = 0;

  for (let window_end = 0; window_end < a.length; window_end++) {
    while (a[window_end] - a[window_start] > 1) {
      window_start++;
    }

    const window_size = window_end - window_start + 1;

    ans = Math.max(window_size, ans);
  }

  return ans;
}

// Time complexity: O(nlog(n)) sorting
// Space compexity: O(1)
